### ISSUE TITLE
[iOS 18] Latest settings dump

### ISSUE THREAD

**Issue by jimmy-zhening-luo** [FifiTheBulldog/ios-settings-urls#72]
iOS settings appears to have a few significant changes. I can tell you which URL schemes appear to work, and which do not (including which schemes seem to map to new setting areas).

I'm happy to volunteer to help with dumping the settings for iOS 18 and building tooling to parse them and auto generate docs, if you're willing to teach me the first part (how to dump settings urls).

**Comment by QkSr4497** [comment]
@jimmy-zhening-luo  - can you please share which scheme in iOS 18 replaced `App-prefs:` ?

**Comment by jimmy-zhening-luo** [comment]
I think it's still App-prefs:. As far as I can tell, root=CASTLE no longer works, you have to use their root=APPLE_ACCOUNT equivalents. Specifically, 'Backup and Restore' no longer worked with *either* castle or apple_account, but they fixed that in iOS 18 Beta 4, where `prefs:root=APPLE_ACCOUNT&path=ICLOUD_SERVICE/BACKUP` once again works.

**Comment by QkSr4497** [comment]
@jimmy-zhening-luo  - I guess you're talking about Shortcuts app.
I'm interested in using URL scheme in my own app.. that seems to be broken or replaced (currently It is only opening settings app without going to the actual path)

```
if let url = URL(string: "App-prefs:APPLE_ACCOUNT") {
            UIApplication.shared.open(url) { accepted in
                print(accepted ? "Success" : "Failure")
            }
}
```

**Comment by colin273** [comment]
@jimmy-zhening-luo The current script in this repo will not work properly above iOS 15 or 16 (can't remember which) due to changes in how localization strings are stored in settings bundles. I'm in the process of rewriting the script to dump URLs on iOS 17, which presumably will work for 18 as well. I'm also taking the opportunity to rewrite it in Python, so it should be much easier to run on a real device; the old version required Node.js. I'll let you know when that's done.

**Comment by jimmy-zhening-luo** [comment]
@colin273, thank you for your work. If/when I can contribute please do let me know!

**Comment by hzlzh** [comment]
> I think it's still App-prefs:. As far as I can tell, root=CASTLE no longer works, you have to use their root=APPLE_ACCOUNT equivalents. Specifically, 'Backup and Restore' no longer worked with _either_ castle or apple_account, but they fixed that in iOS 18 Beta 4, where `prefs:root=APPLE_ACCOUNT&path=ICLOUD_SERVICE/BACKUP` once again works.

Is there still a way to open `App-Prefs:Wallpaper` on iOS 18?

**Comment by vinhquach** [comment]
Hi everyone, I'm trying to find the URL for 'Vocal Shortcuts' only iOS 18. While 'prefs:root=ACCESSIBILITY&path=LIVE_SPEECH_TITLE' works for 'Live Speech' feature, replacing the 'path' parameter with 'VOCAL_SHORTCUTS_TITLE' doesn't. I've tried several similar variations.
Could anyone help?

**Comment by hzlzh** [comment]
> Hi everyone, I'm trying to find the URL for 'Vocal Shortcuts' only iOS 18. While 'prefs:root=ACCESSIBILITY&path=LIVE_SPEECH_TITLE' works for 'Live Speech' feature, replacing the 'path' parameter with 'VOCAL_SHORTCUTS_TITLE' doesn't. I've tried several similar variations. Could anyone help?

What is the command for successful invocation that you mentioned? Is prefs:root still usable?

**Comment by vinhquach** [comment]
> > Hi everyone, I'm trying to find the URL for 'Vocal Shortcuts' only iOS 18. While 'prefs:root=ACCESSIBILITY&path=LIVE_SPEECH_TITLE' works for 'Live Speech' feature, replacing the 'path' parameter with 'VOCAL_SHORTCUTS_TITLE' doesn't. I've tried several similar variations. Could anyone help?
> 
> What is the command for successful invocation that you mentioned? Is prefs:root still usable?

Yes, I opened the URL for 'Live Speech' from Shortcuts. 'LIVE_SPEECH_TITLE' hasn't been documented in the guide yet, but it works.

**Comment by furiosFast** [comment]
have you find a solution?
